Added ability to query within a month for datetimes

hugoh opened the pull request:
With minimal testing on a test library combined with my minimal Chrome extension development knowledge, I _believe_ that this change adds the ability to target library addition date within a month.

simon-weber commented:
Interesting. What kind of playlists are you looking to build with it? Strictly speaking I don't think it gives any more power - since you could instead do `AND(gt(start_of_month), lt(end_of_moth))` - but maybe it makes something substantially easier?

From a UX perspective, it also seems a bit odd to pull out month as the only special unit. Maybe there's a way to make it more generic? Eg switch the condition to "in" and then add a way to declare a duration with the text input?

hugoh commented:
Good points, I hadn't considered have 2 conditions, and I agree on month
being a special case here.

Let me back up: what I'm ultimately interested in doing is generating a
monthly playlist for songs I added to my library. Conceptually, and in a
more generic fashion, I'm thinking about a template as follows:
- Playlist name: "Added in {year}-{month}"
- Condition: added to library during the month of {year}-{month}

As a first step, i thought that manually creating monthly playlists using a
new simpler in-month operator was a good one that I could build upon --
especially since it was my first encounter with the code.

So, maybe my change makes more sense with such a templating feature in
place. That said, that's obviously a larger endeavor as it moves from a 1:1
mapping between definition and playlist to a 1:n between a template
definition and n generated instances.
On Mon, Feb 20, 2017 at 9:37 AM Simon Weber <notifications@github.com>
wrote:

> Interesting. What kind of playlists are you looking to build with it?
> Strictly speaking I don't think it gives any more power - since you could
> instead do AND(gt(start_of_month), lt(end_of_moth)) - but maybe it makes
> something substantially easier?
>
> From a UX perspective, it also seems a bit odd to pull out month as the
> only special unit. Maybe there's a way to make it more generic? Eg switch
> the condition to "in" and then add a way to declare a duration with the
> text input?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/simon-weber/Autoplaylists-for-Google-Music/pull/130#issuecomment-281140389>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAYLIwmGZoOr24B9062YkzaV4wjTiB8oks5rec9ogaJpZM4MF0il>
> .
>

simon-weber commented:
Ah, ok. I can see how that'd be useful, and templating in particular has been here and there. It does seem like a fair amount of work, though.

The closest thing we've got right now is the copy button, which would at least make it somewhat less painful: make the first playlist, copy it a bunch of times, then fill in the months you're interested in.

simon-weber commented:
Though I always feel bad closing PRs, I'm going to since it seems like an AND(gt, lt) fixed your pain point. I'd be interested in merging if we pulled out the time unit somehow (though that's probably a pain because it'd require ui changes).

Feel free to reach out ahead of future PRs: even if you're pretty sure it's something worth merging, I may be able to eg give implementation tips.
